frontend/lint/hacspec_tests/v1-lib/Cargo.toml

```diff
@@ -0,0 +1,9 @@
+[package]
+name = "v1-lib"
+version = "0.1.0"
+edition = "2021"
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+
+[dependencies]
+hacspec-lib = { git = "https://github.com/hacspec/hacspec.git" }
```

Comment:
I know that will be a very sad and concerning news to you Franziskus, but... this is what breaks tests in Nix! Since there is no `Cargo.lock` committed in the Hacspec v1 repo, the dependencies are not known and thus no reproducible build (Nix or not) can be provided :(

Quoting [Cargo doc](https://doc.rust-lang.org/cargo/faq.html#why-do-binaries-have-cargolock-in-version-control-but-not-libraries):
> Users dependent on the library will not inspect the library’s Cargo.lock (even if it exists)

Thus, for non-Nix users, there is no harm in committing a `Cargo.lock`, which will be ignored anyway. So I think we should add `Cargo.lock` in hacspec v1, or move the library to some other repo with `Cargo.lock` committed.

For now, I fixed it on a `with-lockfile` branch.

Reply:
Commiting lock files is not good practice for libraries. So it's nothing that we should do.
I'm afraid nix will need to find a different way to make this work. If it can't handle this, it won't be able to handle most Rust repositories.

Reply:
Oh, my bad. Actually, that was a bug some while ago in `crane`, the Nix helper for building Cargo stuff, but it's fixed now.
So, forget about this, Nix (well, crane) handles that alright.